compiler/rustc_expand/src/mbe/diagnostics.rs

```diff
@@ -245,12 +245,24 @@ pub(super) fn emit_frag_parse_err(
                 e.note(
                     "the macro call doesn't expand to an expression, but it can expand to a statement",
```

Comment:
The note here is correct of the old case, but not for your new code path. It's not that we want to turn an expression into a statement, but a statement into an expression.

Reply:
We are still turning a macro call being an expression (`statement!()` in the `let _ = statement!();`) into a macro call being a statement (`statement!();` in the whole `{...}`), although the whole `{..}` is an expression.

Reply:
oh, right